src/config.js

```diff
@@ -23,6 +23,33 @@
  */
 const env = self.AMP_CONFIG || {};
 
+/**
+ * Check for a custom URL definition in special <meta> tags. Note that this does
+ * not allow for distinct custom URLs in AmpDocShadow instances. The shell is
+ * allowed to define one set of custom URLs via AMP_CONFIG (recommended) or by
+ * including <meta> tags in the shell <head>. Those custom URLs then apply to
+ * all AMP documents loaded in the shell.
+ * @param {string} name
+ * @return {?string}
+ * @private
+ */
+function getMetaUrl(name) {
+  // Avoid exceptions in unit tests
+  if (!self.document || !self.document.head) {
+    return null;
+  }
+
+  const metaEl = self.document.head./*OK*/ querySelector(
+    `meta[name="${name}"]`
+  );
+  return (metaEl && metaEl.getAttribute('content')) || null;
+}
+
+/** @type {!Object<string, ?string>} */
+const metaUrls = {
+  'runtime-host': getMetaUrl('runtime-host'),
```

Comment:
Before merging, we need to confirm that AMP Cache transforms will remove this meta. @Gregable?

Reply:
I pushed a [change](https://github.com/ampproject/amphtml/pull/26829/commits/07873050fb4d29cdaf33e6b16928cd17654c35e3) to only allow `getMetaUrl()` on non-proxy origins. Let me know if you think it is unnecessary and I can back it out.

Reply:
On a separate topic, the `metaUrls` object defined here is not needed. `getMetaUrl('self-host')` can be used directly in the definition of `urls.cdn`:
```
cdn: env['cdnUrl'] || getMetaUrl('runtime-host') || 'https://cdn.ampproject.org',
```